src/Microsoft.ML.Recommender/MatrixFactorizationTrainer.cs

```diff
@@ -86,7 +86,7 @@ namespace Microsoft.ML.Trainers
     /// ]]>
     /// </format>
     /// </example>
-    public sealed class MatrixFactorizationTrainer : TrainerBase<MatrixFactorizationModelParameters>,
+    public sealed class MatrixFactorizationTrainer : ITrainer<MatrixFactorizationModelParameters>,
         IEstimator<MatrixFactorizationPredictionTransformer>
```

Comment:
@wschin do you know why it's not `ITrainerEstimator`  but TrainerBase+ IEstimator?

Reply:
Can't. If you look at `ITrainerEstimator` you'll see that it doesn't produce `ISingleFeaturePredictionTransformer`. That base interface is meant to capture some common cases, but it is not universal.

Reply:
So just going to mark resolved, feel free to reactivate if this angers you! 😄